### PR TITLE
Prepare for 2.0.

### PR DESCRIPTION
**Pull request by dereuromark**
Implements https://github.com/dereuromark/cakephp-tinyauth/issues/37

2.0 should also include/support new auth plugins though.

- [ ] finalize auth plugins support

Ideally also
- [x] proof interfaces/abstractions for getting allow or acl data, e.g. from DB, etc

The migration to a future 3.0 (for Cake4) should then be trivial afterwards.

### CODE REVIEW

File: src/Auth/AclTrait.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ protected function _defaultConfig() {
 			'cacheKey' => 'tiny_auth_acl',
 			'autoClearCache' => null, // Set to true to delete cache automatically in debug mode, keep null for auto-detect
 			'filePath' => null, // Possible to locate INI file at given path e.g. Plugin::configPath('Admin')
-			'file' => 'acl.ini',
+			'file' => 'tinyauth_acl.ini',
```
Comment (saeideng) [inline src/Auth/AclTrait.php]: this is not BC

Comment (dereuromark) [reply src/Auth/AclTrait.php]: thats why the title says "1.x => 2.x major release".

Comment (saeideng) [reply src/Auth/AclTrait.php]: on the `master` ?

Comment (dereuromark) [reply src/Auth/AclTrait.php]: this is not in master, this is in develop.

**Comment by dereuromark** [comment]
@saeideng Are you able to help me finish the auth plugin support? Then we can release this new major :)

**Comment by dereuromark** [comment]
I could still need a hand getting the CakePHP plugins Authentication and Authorization supported.
So that we can release a new major here for Cake3.
And then start upgrading process as another new major for Cake4.

**Comment by dereuromark** [comment]
#### External data integration
With this version a new AllowAdapterInterface has been added to complete an API for other libraries to provide data:
- AllowAdapterInterface
- AclAdapterInterface

Both able to be used to connect this plugin to other libraries for providing allow/ACL input.
It continues to default to the internal INI adapters.
If other adapters - e.g. reading from DB - are used, it will still cache those internally to provide the same general speed it does with internal adapters.

Can someone maybe proof this with a small demo plugin/app? Or help to finalize the 2.0 release?

**Comment by dereuromark** [comment]
I think I am ready now to release v2
I checked this with the new https://github.com/dereuromark/cakephp-tinyauth-backend/ backend implementation and all works fine.

The array structure is fine for now IMO.

**Pull request merged**